tests/jest/application-navigation/ApplicationNavigation.test.jsx

```diff
@@ -0,0 +1,82 @@
+import React from 'react';
+
+import ApplicationNavigation from '../../../src/application-navigation/ApplicationNavigation';
+
+describe('ApplicationNavigation', () => {
+  it('should render default element', () => {
+    const shallowComponent = shallow(
+      <ApplicationNavigation />,
+    );
+
+    expect(shallowComponent).toMatchSnapshot();
+  });
+
+  it('should render with all props', () => {
+    const shallowComponent = shallow(
+      <ApplicationNavigation
+        activeBreakpoint="large"
+        titleConfig={{
+          title: 'test-title',
+        }}
+        activeNavigationItemKey="my-test-key"
+        hero={<div>my test hero</div>}
+        userConfig={{
+          name: 'user-name',
+          detail: 'user-detail',
+          initials: 'user-initials',
+          imageSrc: 'user-src',
+        }}
+        navigationItems={[{ text: 'test-text', key: 'my-test-key' }]}
+        extensionItems={[{ text: 'test-text-1', key: 'my-test-key-1', icon: <span>my icon</span> }]}
+        utilityItems={[{ text: 'test-text-2', key: 'my-test-key-2', icon: <span>my icon</span> }]}
+        notifications={{ 'my-test-key-1': 3 }}
+        navigationRenderFunction={jest.fn()}
+        onSelectNavigationItem={jest.fn()}
+        onSelectExtensionItem={jest.fn()}
+        onSelectUtilityItem={jest.fn()}
+        onSelectSettings={jest.fn()}
```

Comment:
Did we want to ensure the navigation prompting was being handled and/or bypassed as expected since we wrap the `onSelectNavigationItem` and `onSelectLogout` functions that consumers will be providing?

Reply:
Those workflows are covered by the webdriver tests.

Reply:
We could test them here, but they'd be...uglier.

Reply:
Yeah I'd leave it as it. I didn't see the wdio tests till later on